metron-platform/metron-test-utilities/src/main/java/org/apache/metron/test/mock/MockHTable.java

```diff
@@ -437,15 +463,14 @@ public ResultScanner getScanner(byte[] family, byte[] qualifier)
     return getScanner(scan);
   }
 
-  List<Put> putLog = new ArrayList<>();
-
   public List<Put> getPutLog() {
-    return putLog;
+    return ImmutableList.copyOf(putLog.get());
   }
 
   @Override
   public void put(Put put) throws IOException {
-    putLog.add(put);
+    putLog.get().add(put);
```

Comment:
To @justinleet's point, while the `Supplier.get()` call is thread-safe, the resulting list is not.  I would recommend adjusting the construction of the `ArrayList<Put>` to `Collections.synchronizedList(new ArrayList<>())` to ensure that the resulting `add` call is happening in a thread-safe way.  At that point, it's unclear why we would use the supplier at all.  

On the whole, it's interesting that this may be at the root of the unit test weirdness.  Have you gotten the tests to fail in a somewhat repeatable manner and does this change correct it?

Standard disclaimer applies: all of this is based on my understanding of Java, which can be obviously incomplete, so let me know if I got something wrong here.